ui/app/components/pages/confirm-transaction-base/confirm-transaction-base.component.js

```diff
@@ -0,0 +1,375 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import ConfirmPageContainer, { ConfirmDetailRow } from '../../confirm-page-container'
+import { formatCurrency, getHexGasTotal } from '../../../helpers/confirm-transaction/util'
+import { isBalanceSufficient } from '../../send_/send.utils'
+import { DEFAULT_ROUTE } from '../../../routes'
+import { conversionGreaterThan } from '../../../conversion-util'
+import { MIN_GAS_LIMIT_DEC } from '../../send_/send.constants'
+
+export default class ConfirmTransactionBase extends Component {
+  static contextTypes = {
+    t: PropTypes.func,
+  }
+
+  static propTypes = {
+    match: PropTypes.object,
+    history: PropTypes.object,
+    // Redux props
+    txData: PropTypes.object,
+    tokenData: PropTypes.object,
+    methodData: PropTypes.object,
+    tokenProps: PropTypes.object,
+    isTxReprice: PropTypes.bool,
+    nonce: PropTypes.string,
+    fromName: PropTypes.string,
+    fromAddress: PropTypes.string,
+    toName: PropTypes.string,
+    toAddress: PropTypes.string,
+    transactionStatus: PropTypes.string,
+    ethTransactionAmount: PropTypes.string,
+    ethTransactionFee: PropTypes.string,
+    ethTransactionTotal: PropTypes.string,
+    fiatTransactionAmount: PropTypes.string,
+    fiatTransactionFee: PropTypes.string,
+    fiatTransactionTotal: PropTypes.string,
+    hexGasTotal: PropTypes.string,
+    balance: PropTypes.string,
+    currentCurrency: PropTypes.string,
+    conversionRate: PropTypes.number,
+    clearConfirmTransaction: PropTypes.func,
+    cancelTransaction: PropTypes.func,
+    clearSend: PropTypes.func,
+    sendTransaction: PropTypes.func,
+    editTransaction: PropTypes.func,
+    showCustomizeGasModal: PropTypes.func,
+    updateGasAndCalculate: PropTypes.func,
+    showTransactionConfirmedModal: PropTypes.func,
+    // Component props
+    action: PropTypes.string,
+    hideDetails: PropTypes.bool,
+    hideData: PropTypes.bool,
+    detailsComponent: PropTypes.node,
+    dataComponent: PropTypes.node,
+    summaryComponent: PropTypes.node,
+    contentComponent: PropTypes.node,
+    title: PropTypes.string,
+    subtitle: PropTypes.string,
+    hideSubtitle: PropTypes.bool,
+    valid: PropTypes.bool,
+    errorMessage: PropTypes.string,
+    warning: PropTypes.string,
+    identiconAddress: PropTypes.string,
+    onEdit: PropTypes.func,
+    onEditGas: PropTypes.func,
+    onCancel: PropTypes.func,
+    onSubmit: PropTypes.func,
+  }
+
+  componentDidUpdate () {
+    const {
+      transactionStatus,
+      showTransactionConfirmedModal,
+      history,
+      clearConfirmTransaction,
+    } = this.props
+
+    if (transactionStatus === 'dropped') {
+      showTransactionConfirmedModal({
+        onHide: () => {
+          clearConfirmTransaction()
+          history.push(DEFAULT_ROUTE)
+        },
+      })
+
+      return
+    }
+  }
+
```

Comment:
What are your thoughts on moving some or all of the logic in `getError` and `validateEditGas` to either reducers or utils?

Reply:
I think keeping errors in the redux state makes sense when there's an app-wide display for errors (like how GitHub has it under the nav bar) but when the error is localized to a component then I don't think it's necessary, since it's basically up to the component how the error should be handled (displayed somewhere, disable buttons, etc.).

As for utils, I get that by doing so it keeps the component file smaller, but in my opinion the purpose of a util is re-usability. I want to avoid passing in a bunch of arguments into a util function as that just makes the util very specific and hard to use. In this case, I think `isBalanceSufficient` is great as a util, but `getError` would be too specific to the component. `validateEditGas` could probably be moved to the container file, but we should figure out how to deal with the translation of error messages.